src/ITest/java/uk/gov/companieshouse/pscstatementdataapi/steps/PscStatementsSteps.java

```diff
@@ -0,0 +1,274 @@
+        String uri = "/company/{company_number}/persons-with-significant-control-statements/{statement_id}/internal";
+
+        HttpHeaders headers = new HttpHeaders();
+        CucumberContext.CONTEXT.set("contextId", "5234234234");
+        headers.set("x-request-id", CucumberContext.CONTEXT.get("contextId"));
+        headers.set("ERIC-Identity", "TEST-IDENTITY");
+        headers.set("ERIC-Identity-Type", "KEY");
+        headers.set("ERIC-Authorised-Key-Privileges", "internal-app");
+
+        HttpEntity<String> request = new HttpEntity<String>(null, headers);
+
+        ResponseEntity<Void> response = restTemplate.exchange(uri, HttpMethod.DELETE, request, Void.class, companyNumber, statementId);
+
+        CucumberContext.CONTEXT.set("statusCode", response.getStatusCodeValue());
+    }
+
+    @When("psc statement id does not exist for {string}")
+    public void statement_does_not_exist(String statementId) {
+        Assertions.assertThat(pscStatementRepository.existsById(statementId)).isFalse();
+    }
+
+    @When("CHS kafka API service is unavailable")
+    public void chs_kafka_service_unavailable() throws IOException {
+        doThrow(ServiceUnavailableException.class)
+                .when(pscStatementApiService).invokeChsKafkaApiWithDeleteEvent(any(), any(), any(), any());
+    }
+
+    @Then("the CHS Kafka API is not invoked")
+    public void chs_kafka_api_not_invoked() throws IOException {
+        verify(pscStatementApiService, times(0)).invokeChsKafkaApi(any(), any(), any());
+    }
+
+    @Then("the CHS Kafka API delete is invoked for company number {string} with statement id {string} and the correct statement data")
+    public void chs_kafka_api_invoked_delete(String companyNumber, String statementId) throws IOException {
+        File statementFile = new ClassPathResource("/json/output/psc_statement.json").getFile();
+        Statement pscStatement = objectMapper.readValue(statementFile, Statement.class);
+        verify(pscStatementApiService).invokeChsKafkaApiWithDeleteEvent("5234234234", companyNumber, statementId, pscStatement);
+    }
+
+    @When("a statement exists with id {string}")
+    public void statement_exists(String statementId) {
+        Assertions.assertThat(pscStatementRepository.existsById(statementId)).isTrue();
+    }
+
+    @Then("the CHS Kafka API is invoked for company number {string} with statement id {string}")
+    public void chs_kafka_api_invoked(String companyNumber, String statementId) {
+        verify(pscStatementApiService).invokeChsKafkaApi("5234234234", companyNumber, statementId);
+    }
+
+    @Then("nothing is persisted in the database")
+    public void nothing_persisted_to_database() {
+        List<PscStatementDocument> pscDocs = pscStatementRepository.findAll();
+        Assertions.assertThat(pscDocs).hasSize(0);
+    }
+
+    @After
+    public void dbStop(){
+        mongoDBContainer.stop();
+    }
+}
```

Comment:
Like this as a whole although it is a bit of a monster file. Totally optional but we could split off situation, action and confirmation steps and have a given, when and then file or something. Maybe one to look at again when more tests are added

Reply:
I did think about it but the majority of the steps are used in multiple files so I think we end up with a common steps that is almost as big

Reply:
And then have to move steps between files if we add more tests